File: ebook/src/services/cartService.js
import config from 'config' 
import {postRequest, postRequest_v2} from '../util/ajax'

export const getCart = (id, callback) => {
    const data = {id:id};
    console.log(data);
    const url = `${config.apiUrl}/getCart`;
    postRequest_v2(url, data, callback);
}

export const addCart = (data, callback) => {
    const url = `${config.apiUrl}/addCart`;
    console.log(data);
    postRequest(url, data, callback);
}

export const deleteCart = (data, callback) => {
    const url = `${config.apiUrl}/deleteCart`;
    console.log(data);
    postRequest(url, data, callback);
}

File: ebook_backend/src/main/java/com/example/demo/service/CartService.java
package com.example.demo.service;

import com.example.demo.entity.Cart;
import com.example.demo.entity.CartResponse;
import com.example.demo.utils.msgutils.Msg;

import java.util.List;
import java.util.Map;

public interface CartService {

    public List<CartResponse> getCart(Integer id);

    public Msg addCart(Cart data);

    public Msg deleteCart(Map<String, String> params);
}
